Add vertical and horizontal scrollbars in collectionview docs. (#982)
telerik/maui-docs: controls/entry/visual-states.md

---
title: Visual States
page_title: .NET MAUI Entry Documentation - Visual States
description: Learn how to set the border color, background color and other in the different visual states of the Telerik Entry for .NET MAUI.
position: 9
tags: .net maui, telerik .net maui, ui for .net maui, entry, states, microsoft .net maui
slug: entry-visual-states
---

# .NET MAUI Entry Visual States

The Entry provides the following `CommonStates` visual states:

| Visual State | Description |
| ------------- | --------------- |

> | `Normal` | Applies when the entry is in normal state. |

Reviewer: Replace "entry" with "Entry" in all descriptions.

| `Focused` | Applies when the entry is focused. |
| `MouseOver` | (Desktop-only) Applies when the mouse pointer is over the entry. |
| `Invalid` | Applies when there is a validation and the text entered inside the entry is not valid (the `IsValueValid` property is set to `false`). |
| `InvalidFocused` | Applies when the entry is focused and there is a validation and the text entered inside the entry is not valid (the `IsValueValid` property is set to `false`). |
| `InvalidMouseOver` | (Desktop-only) Applies when the mouse is over the entry and there is a validation and the text entered inside the entry is not valid (the `IsValueValid` property is set to `false`). |
| `ReadOnly` | Applies when the entry is not editable (the `IsReadOnly` property is set to `false`). |
| `ReadOnlyFocused` | Applies when the entry is focused and not editable (the `IsReadOnly` property is set to `false`). |

> | `ReadOnlyMouseOver` | (Desktop-only) Applies when the mouse pointer is over the entry, and the entry is not editable (the `IsReadOnly` property is set to `false`) |

Reviewer: ```suggestion
| `ReadOnlyMouseOver` | (Desktop-only) Applies when the mouse pointer is over the entry, and the entry is not editable (the `IsReadOnly` property is set to `false`). |
```

| `ReadOnlyInvalid` | Applies when the entry is not editable (the `IsReadOnly` property is set to `false`) and not valid (the `IsValueValid` property is set to `false`). |

> | `ReadOnlyInvalidFocused` | Applies when the entry is focused, not editable (the `IsReadOnly` property is set to `false`) and not valid (the `IsValueValid` property is set to `false`). |

Reviewer: ```suggestion
| `ReadOnlyInvalidFocused` | Applies when the entry is focused, not editable (the `IsReadOnly` property is set to `false`), and not valid (the `IsValueValid` property is set to `false`). |
```

> | `ReadOnlyInvalidMouseOver` | (Desktop-only) Applies when the mouse pointer is over the entry, and the entry is not editable (the `IsReadOnly` property is set to `false`) and not valid (the `IsValueValid` property is set to `false`). |

Reviewer: ```suggestion
| `ReadOnlyInvalidMouseOver` | (Desktop-only) Applies when the mouse pointer is over the entry, the entry is not editable (the `IsReadOnly` property is set to `false`), and not valid (the `IsValueValid` property is set to `false`). |
```

| `Disabled` | Applies when the entry is disabled. |

> ### Using the Visual States

Reviewer: ```suggestion
## Using the Visual States
```

> The following example demonstrates how to use the Entry Visual States.

Reviewer: ```suggestion
The following example demonstrates how to use the Entry's Visual States.
```

**1.** Define the Entry in XAML:

<snippet id='entry-style' />

> **2.** Define the entry style in the page's resources:

Reviewer: ```suggestion
**2.** Define the Entry's style in the page's resources:
```

<snippet id='entry-styling' />

> **2.** Define the clear button style in the page's resources:

Reviewer: Fix the numbering for this step and all that follow.

<snippet id='entry-styling-clear-button' />

**2.** Define the validation error label style in the page's resources:

<snippet id='entry-styling-validation-error-label' />

**2.** Define the validation error image style in the page's resources:

<snippet id='entry-styling-validation-error-image' />

**3.** Add the `telerik` namespace:

```XAML
xmlns:telerik="http://schemas.telerik.com/2022/xaml/maui"
```

This is the result on WinUI: 

![.NET MAUI Entry Visual States](images/)

> >tip For a runnable example demonstrating the Entry visual state, see the [SDKBrowser Demo Application]({%slug sdkbrowser-app%}) and go to the **Entry > Styling** category.

Reviewer: ```suggestion
>tip For a runnable example demonstrating the Entry's Visual States, see the [SDKBrowser Demo Application]({%slug sdkbrowser-app%}) and go to the **Entry > Styling** category.
```

---
title: Visual States
page_title: .NET MAUI Entry Documentation - Visual States
description: Learn how to set the border color, background color, and other visual states for the Telerik UI for .NET MAUI Entry control.
position: 9
tags: .net maui, telerik .net maui, ui for .net maui, entry, states, microsoft .net maui
slug: entry-visual-states
---

# .NET MAUI Entry Visual States

> This article describes the visual states the Entry provides. 

Reviewer: ```suggestion
This article describes the visual states provided by the Entry control. You can use these visual states to change the appearance of the control based on its state—such as when it’s disabled, focused, hovered, and more.
```

> You can use the visual states to change the visual appearance of the control depending on the state the control is set&mdash;whether it's disabled, or focused, or has the mouse pointer over and more.

Reviewer: ```suggestion
```
